src/nbd.c

```diff
@@ -951,6 +951,10 @@ gboolean r_nbd_run_server(gint sock, GError **error)
 
 	res = TRUE;
 out:
+	g_clear_pointer(&ctx.url, g_free);
```

Comment:
That's quite close to justifying a cleanup method for `RaucNBDContext` :wink: 
But fine first of all.